### DISCUSSION TITLE
Issue evaluating Temperature from Pressure and Enthalpy

### DISCUSSION THREAD

**Discussion by LugPaw** [usnistgov/REFPROP-issues#615] [General]
In an Excel/VBA simulation that I was running, REFPROP complained about a “type mismatch” trying to extract the temperature of WATER from the given pressure and enthalpy input in the shaded row of the table below. 

This is weird because the inputs  are nowhere near edge of the range of validity. In fact, perturbing the enthalpy by 0.5 and 1 kJ/kg on the minus side gives the correct values, while I have to perturb it by a full 1 kJ/kg to get it to evaluate correctly on the plus side.

Is there a patch that I need to download to fix this issue?

Thanks
![image](https://github.com/usnistgov/REFPROP-issues/assets/146107943/a972225a-25cc-4914-aeb9-7e56c89213f2)

**Comment by ianhbell** [comment]
Yes, this is a bug, and we have a beta version that fixes this bug, but it has some other regressions (new bugs) so I cannot recommend the beta without reservation. If you want to try the beta, please email me at: ian.bell@nist.gov

**Comment by ianhbell** [comment]
Mysteriously your bug does not appear to be present on my mac with REFPROP 10.0. On my machine:

``` python
import os 
import ctREFPROP.ctREFPROP as ct
root = os.getenv('RPPREFIX')
RP = ct.REFPROPFunctionLibrary(root)
RP.SETPATHdll(root)
RP.SETFLUIDSdll('WATER')

p = 21.82990e6 # [Pa]
for h_kJkg in [137.5115, 138.0115, 138.5115, 139.0115, 139.5115]:
    r = RP.REFPROPdll('','HP', 'T', RP.MASS_BASE_SI,0,0, h_kJkg*1e3, p, [1.0])
    print(r.Output[0]-273.15)
```
yields no failures
```
28.06610446422252
28.1873315134373
28.308564081043983
28.429795268751946
28.551025071499396
```

**Comment by LugPaw** [comment]
I wonder if it has something to do with REFPROP 10 and Windows.

One of my colleagues has REFPROP 10 and has the same issue whereas the other who uses REFPROP 9 (NIST-Steam) does not.

I was able to circumvent the Temperature issue by using an external iteration, but noticed that density, viscosity, and other thermophysical properties also cannot be evaluated by REFPROP at this particular P,h combination.

Any suggestions as to a robust workaround?
